fe/src/main/java/org/apache/doris/analysis/AddMaterializedViewClause.java

```diff
@@ -0,0 +1,253 @@
+                }
+                else if (functionChild0 instanceof CastExpr
+                        && (functionChild0.getChild(0) instanceof SlotRef)) {
+                    slotRef = (SlotRef) functionChild0.getChild(0);
+                } else {
+                    throw new AnalysisException("The children of aggregate function only support one original column. "
+                                                        + "Error function: " + functionCallExpr.toSqlImpl());
+                }
+                meetAggregate = true;
+                String columnName = slotRef.getColumnName().toLowerCase();
+                if (!mvColumnNameSet.add(columnName)) {
+                    ErrorReport.reportAnalysisException(ErrorCode.ERR_DUP_FIELDNAME, columnName);
+                }
+                MVColumnItem mvColumnItem = new MVColumnItem(columnName);
+                mvColumnItem.setAggregationType(AggregateType.valueOf(functionName.toUpperCase()), false);
+                mvColumnItemList.add(mvColumnItem);
+            }
+        }
+        // TODO(ML): only value columns of materialized view, such as select sum(v1) from table
+    }
+
+    private void analyzeFromClause() throws AnalysisException {
+        List<TableRef> tableRefList = selectStmt.getTableRefs();
+        if (tableRefList.size() != 1) {
+            throw new AnalysisException("The materialized view only support one table in from clause.");
+        }
+        baseIndexName = tableRefList.get(0).getName().getTbl();
+    }
+
+    private void analyzeOrderByClause() throws AnalysisException {
+        if (selectStmt.getOrderByElements() == null) {
+            /**
+             * Supplement keys of MV columns
+             * For example: select k1, k2 ... kn , sum(v1), sum(v2) from t1
+             * The default key columns are first 36 bytes of the columns in define order.
+             * If the number of columns in the first 36 is less than 3, the first 3 columns will be used.
+             * column: k1, k2, k3... km. The key is true.
+             * Supplement non-key and non-value of MV columns
+             * column: km... kn. The key is false, aggregation type is none, isAggregationTypeImplicit is true.
+             */
+            int keyStorageLayoutBytes = 0;
+            for (int i = 0; i < selectStmt.getResultExprs().size(); i++) {
+                MVColumnItem mvColumnItem = mvColumnItemList.get(i);
+                Expr resultColumn = selectStmt.getResultExprs().get(i);
+                if (mvColumnItem.getAggregationType() != null) {
+                    break;
+                }
+                keyStorageLayoutBytes += resultColumn.getType().getStorageLayoutBytes();
+                if ((i + 1) <= Config.DEFAULT_DUP_KEYS_COUNT || keyStorageLayoutBytes < Config.DEFAULT_DUP_KEYS_BYTES) {
+                    mvColumnItem.setIsKey(true);
+                } else {
+                    mvColumnItem.setAggregationType(AggregateType.NONE, true);
```

Comment:
If user specified aggregate method for column, all other columns should be key columns,
or there may cause some problem when executing on BE.